### PR TITLE
added logger warning and error for cosine similarity computation for …

### PR DESCRIPTION
**Pull request by vivekpandit-msft**
Added logger warning and error for cosine similarity computation for zero vectors

### CODE REVIEW

**Comment by alexchaomander** [comment]
This PR addresses #130

**Review by alexchaomander — DISMISSED** [review]
LGTM!

**Review by alexchaomander — APPROVED** [review]

File: python/tests/test_unit_cosine_similarity.py
Comment (jjhenkel) [inline python/tests/test_unit_cosine_similarity.py:8]: Hey @vivekpandit-msft, are you able to run tests here? Have you done a `poetry install && poetry shell` and `pytest tests/`? 

When I do, I get the following: 
```
...
PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.
  You need to install a suitable plugin for your async framework, for example:
  ...
```

I'm pretty sure this warning is coming from not marking these tests as `async` (you made the functions `async`, which was a necessary step, but we also need to tell `pytest` this info)

File: python/tests/test_unit_cosine_similarity.py
Comment (jjhenkel) [inline python/tests/test_unit_cosine_similarity.py:21]: To fix the warning from `pytest`, you'll need something like this (see suggestion). Don't forget to update the relevant import too.

```suggestion
@mark.asyncio
async def test_cosine_similarity_zero_query():
```

File: python/semantic_kernel/memory/volatile_memory_store.py
Comment (jjhenkel) [inline python/semantic_kernel/memory/volatile_memory_store.py:53]: I don't think this needs to be `async` (you aren't `await`ing anything in here)

File: python/semantic_kernel/memory/volatile_memory_store.py
Comment (jjhenkel) [inline python/semantic_kernel/memory/volatile_memory_store.py:37]: If `compute_similarity_scores` is `async` we should `await` it here, but I'd rather you just make it non-async

**Review by jjhenkel — CHANGES_REQUESTED** [review]
Thanks for this @vivekpandit-msft! I left a few requests for fixes. Some other broader requests:

Can you run `make pre-commit` before and apply fixes, looks like some things are out-of-compliance w/ the linter. This also runs things like `isort` (which we will be adding to the linting workflow soon).

My output from `make pre-commit` on this PR:
```
make pre-commit
poetry run pre-commit run --all-files -c .conf/.pre-commit-config.yaml
Check Toml...............................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Mixed line ending........................................................Passed
black....................................................................Passed
isort....................................................................Failed
- hook id: isort
- files were modified by this hook

Fixing /home/jjhenkel/workspace/semantic-kernel/python/semantic_kernel/core_skills/__init__.py
Fixing /home/jjhenkel/workspace/semantic-kernel/python/semantic_kernel/core_skills/file_io_skill.py
Fixing /home/jjhenkel/workspace/semantic-kernel/python/tests/test_time_skill.py
Fixing /home/jjhenkel/workspace/semantic-kernel/python/semantic_kernel/memory/volatile_memory_store.py

flake8...................................................................Failed
- hook id: flake8
- exit code: 1

python/semantic_kernel/memory/volatile_memory_store.py:57:89: E501 line too long (110 > 88 characters)
python/semantic_kernel/memory/volatile_memory_store.py:58:89: E501 line too long (120 > 88 characters)
python/semantic_kernel/memory/volatile_memory_store.py:65:89: E501 line too long (112 > 88 characters)
python/semantic_kernel/memory/volatile_memory_store.py:89:89: E501 line too long (92 > 88 characters)
```

Can you structure the tests similar to the others that were recently merged (put them under `./unit/memory/test_volatile_memory_store.py`)?

File: python/tests/test_unit_cosine_similarity.py
Comment (vivekpandit-msft) [reply python/tests/test_unit_cosine_similarity.py:21]: Got it. Done

File: python/semantic_kernel/memory/volatile_memory_store.py
Comment (vivekpandit-msft) [reply python/semantic_kernel/memory/volatile_memory_store.py:53]: Done

File: python/semantic_kernel/memory/volatile_memory_store.py
Comment (vivekpandit-msft) [reply python/semantic_kernel/memory/volatile_memory_store.py:37]: Done

File: python/tests/test_unit_cosine_similarity.py
Comment (vivekpandit-msft) [reply python/tests/test_unit_cosine_similarity.py:8]: Yes, after all your suggested changes it works

**Comment by vivekpandit-msft** [comment]
> Thanks for this @vivekpandit-msft! I left a few requests for fixes. Some other broader requests:
> 
> Can you run `make pre-commit` before and apply fixes, looks like some things are out-of-compliance w/ the linter. This also runs things like `isort` (which we will be adding to the linting workflow soon).
> 
> My output from `make pre-commit` on this PR:
> 
> ```
> make pre-commit
> poetry run pre-commit run --all-files -c .conf/.pre-commit-config.yaml
> Check Toml...............................................................Passed
> Check Yaml...............................................................Passed
> Fix End of Files.........................................................Passed
> Mixed line ending........................................................Passed
> black....................................................................Passed
> isort....................................................................Failed
> - hook id: isort
> - files were modified by this hook
> 
> Fixing /home/jjhenkel/workspace/semantic-kernel/python/semantic_kernel/core_skills/__init__.py
> Fixing /home/jjhenkel/workspace/semantic-kernel/python/semantic_kernel/core_skills/file_io_skill.py
> Fixing /home/jjhenkel/workspace/semantic-kernel/python/tests/test_time_skill.py
> Fixing /home/jjhenkel/workspace/semantic-kernel/python/semantic_kernel/memory/volatile_memory_store.py
> 
> flake8...................................................................Failed
> - hook id: flake8
> - exit code: 1
> 
> python/semantic_kernel/memory/volatile_memory_store.py:57:89: E501 line too long (110 > 88 characters)
> python/semantic_kernel/memory/volatile_memory_store.py:58:89: E501 line too long (120 > 88 characters)
> python/semantic_kernel/memory/volatile_memory_store.py:65:89: E501 line too long (112 > 88 characters)
> python/semantic_kernel/memory/volatile_memory_store.py:89:89: E501 line too long (92 > 88 characters)
> ```
> 
> Can you structure the tests similar to the others that were recently merged (put them under `./unit/memory/test_volatile_memory_store.py`)?

I have made all the changes you suggested @jjhenkel. Thanks for the review.

**Comment by dluc** [comment]
@vivekpandit-msft could address the conflict. @jjhenkel I'll wait for your approval, thanks!

File: python/semantic_kernel/memory/volatile_memory_store.py
Comment (jjhenkel) [inline python/semantic_kernel/memory/volatile_memory_store.py:57]: Nit: we should all discuss the format we want docstrings to be in. 

I've been using Copilot's suggestions to help me generate docstrings, but I've noticed that I haven't been consistent (sometimes it's `:param x: blah` style; other times, it's `Args:\n...` style) 

Let's all discuss and make sure docstrings are consistent at some point in the near future.

**Review by jjhenkel — DISMISSED** [review]
Thanks for the changes @vivekpandit-msft! LGTM now, left some minor comments (but mostly things we should just follow-up on, and things that don't apply just to this PR like docstring consistency).

File: python/tests/unit/memory/test_volatile_memory_store.py
Comment (vivekpandit-msft) [reply python/tests/unit/memory/test_volatile_memory_store.py:2]: Done

**Comment by vivekpandit-msft** [comment]
> @vivekpandit-msft could address the conflict. @jjhenkel I'll wait for your approval, thanks!

Addressed the conflict and @jjhenkel's comments

**Review by dluc — APPROVED** [review]

**Pull request merged**
